tests/run/break-opt.scala

```diff
@@ -0,0 +1,72 @@
+import scala.util.*
+
+object breakOpt:
```

Comment:
With `run` tests we only check the semantics of the `boundary`/`break`. We also need to test that the optimization in happening. I propose using bytecode tests similar to https://github.com/lampepfl/dotty/blob/main/compiler/test/dotty/tools/backend/jvm/DottyBytecodeTests.scala#L1519-L1558. There we will be able to check that the gotos and returns are generated as expected.

Reply:
I could add these tests.

Reply:
Yes, please go ahead!

Reply:
Can cherrypick 871576691517f119f8cbe8513e333d5db0a61a68

Reply:
These will have to be adapted to latest codegen. In fact, in the interest of robustness, it's probably best to just test that a test contains a labeled block or a try or both. That's all we need to know, and that leaves open later changes in code generation.

Reply:
Updated. Now it can be cherry-picked from 5c7a0002ffc68f89ba1ca26d04dd4cb3d73e77ec.